src/photos/queries/queries.js

```diff
@@ -44,6 +41,19 @@ export const buildTimelineQuery = () => ({
   }
 })
 
+export const buildPhotosTrashed = () => ({
+  definition: Q(DOCTYPE_FILES)
+    .where({
+      class: 'image',
+      trashed: true
+    })
+    .indexFields(['class', 'trashed']),
```

Comment:
since the predicate is static, should we use a partial index @paultranvan ?

Reply:
Absolutely: using
```
.partialIndex({
      class: 'image',
      trashed: true
    })
 ```
 will pre-filter documents entering the index, so here, only the trashed images. While using the `.where` will index all the docs, making it unnecessarily bigger, and slower.

Reply:
Thanks for confirming Quentin's explanation. In the case of referenced_by, it doesn't seem to work because this field isn't part of the `io.cozy.files` document, but I may be wrong.